Entity

MireilleNyoubissi opened the pull request.

MireilleNyoubissi committed:
Added Bookingmanager entity

MireilleNyoubissi committed:
Updated BookingManager field in the Booking class

MireilleNyoubissi committed:
Created Report entity

MireilleNyoubissi committed:
Updated Booking class with report field

MireilleNyoubissi committed:
Added Customer field in Report class

MireilleNyoubissi committed:
Updated Bookingmanager field with report field

MireilleNyoubissi committed:
updated Report class with bookingManager field

MireilleNyoubissi committed:
Created HR entity

MireilleNyoubissi committed:
Upadated Report class with Hr field

MireilleNyoubissi committed:
Upadated HR class with customer field

MireilleNyoubissi committed:
Created Ticket entity

MireilleNyoubissi committed:
Upadated Customer class with tickets field

MireilleNyoubissi committed:
Upadated Customer field in Booking class with nullable false

MireilleNyoubissi committed:
Created Person entity

chimaihueze reviewed (changes requested):
**Requested Changes**
- Staff should inherit from Person
- Customer and HR should inherit from Staff

**NB:** These layers of inheritance means individual classes shouldn't have duplicate code, i.e., id, name, etc.

- First and last name fields should be camel-case.
- The _reports_ field on _Booking_ class, has a Set data type, this should be a List.

MireilleNyoubissi committed:
Modified firstName and lastName attribute to meet CamelCase naming conventions

MireilleNyoubissi committed:
Changed Set data type to List on reports field

MireilleNyoubissi committed:
Created Staffs entity

MireilleNyoubissi committed:
Modified BookingManager class to extends staffs

MireilleNyoubissi committed:
Modified HR class to extends staffs

MireilleNyoubissi committed:
Modified Customer class to extends staffs

MireilleNyoubissi committed:
Removed s from Staff

MireilleNyoubissi committed:
Removed s from Staff

MireilleNyoubissi committed:
Removed s from Staff

chimaihueze reviewed (changes requested):
### Requested Changes

- The **Customer, Staff, HR, BookingManager** classes should not have the _id_ field since the **Person** class already have this field, and they inherit from it. Having _id_ on these classes is code repetition. 
- **HR** class should extend _Staff_, and not _Staffs_.
- The _Reports_ field on **HR** class should be all lowercase.
- The _Reports_ and _Bookings_ field in **BookingManager** class, should be all lowercase.
- The _tickets_ field in **Customer** class should be of data type _List_, not _Set_.
- Refactor **Person** class to **User**. It sounds better than Person.
- **Staff** class should have these fields: _department_, _salary_.
- **Staff** class is named Staffs in your code, refactor to Staff.
- Remove the _Customer_name_ field in the **Customer** class, it is redundant. It already inherits firstName, lastName

MireilleNyoubissi committed:
Refactored Person class to User

MireilleNyoubissi committed:
Changed Set data type to list on the ticket field
removed Id field,
removed customer_name field

MireilleNyoubissi committed:
Removed Id field, added department and salary field

MireilleNyoubissi committed:
Removed Id field, Changed reports and booking fields to lowercase

MireilleNyoubissi committed:
Removed Id field, changed reports filed to lowercase

chimaihueze reviewed (changes requested):
### **Requested Changes**

- User is not an entity, but a super class that Customer, HR, BookingManager entities are created from. Replace Entity annotation with MappedSuperClass.

- Remove the @ Column annotation from all fields on the User and Staff class.

- Staff is not an entity - remove the @ Entity annotation. DO NOT ADD MappedSuperClass.

- Add @ NoArgsConstructor and @ AllArgsConstructor to all classes including the User and Staff class.

- Build these changes, make sure it runs before requesting for another pull request.

MireilleNyoubissi committed:
Removed @Column and @Entity annotations on User class,
Added @MappedSuperClass, @NoAgrsConstructor, @AllArgsConstructor annotation to User class

MireilleNyoubissi committed:
Removed @Entity annotations on Staff class,
Added @NoAgrsConstructor, @AllArgsConstructor

MireilleNyoubissi committed:
Added @NoAgrsConstructor, @AllArgsConstructor to Booking class

MireilleNyoubissi committed:
Added @NoAgrsConstructor, @AllArgsConstructor to BookingManager class

MireilleNyoubissi committed:
Added @NoAgrsConstructor, @AllArgsConstructor to Customer class

MireilleNyoubissi committed:
Added @NoAgrsConstructor, @AllArgsConstructor to HR class

MireilleNyoubissi committed:
Added @NoAgrsConstructor, @AllArgsConstructor to Payment class

MireilleNyoubissi committed:
Added @NoAgrsConstructor, @AllArgsConstructor to Report class

MireilleNyoubissi committed:
Added @NoAgrsConstructor, @AllArgsConstructor to Room class

MireilleNyoubissi committed:
Added @NoAgrsConstructor, @AllArgsConstructor to Ticket class

MireilleNyoubissi committed:
Removed @Table to User class

MireilleNyoubissi committed:
Added implements Serializable to Customer class

chimaihueze commented:
changes approved
